service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -215,6 +215,39 @@ public String getOrCreateUserManagedIdentityForUser(
     }
   }
 
+  /**
+   * Returns the 'Action Managed Identity' that the requesting user has access to, if one exists.
+   *
+   * @param billingProfileId By convention, this billingProfileId is equivalent to the Sam
+   *     resourceId. Also known as the 'SpendProfileId' in this codebase, when interacting with
+   *     Azure workspaces. It is used to uniquely identify an identity that may grant users access
+   *     to private Azure Container Registries.
+   * @param request The request from user asking for the identity. Used to impersonate the user when
+   *     querying Sam.
+   * @return Display name of the Action Identity, if one exists. Might look something like
+   *     "586d120b-c4c-pull_image"
+   */
+  public Optional<String> getActionIdentityForUser(
+      String billingProfileId, AuthenticatedUserRequest request) throws InterruptedException {
+
+    String token = getSamUser(request).getBearerToken().getToken();
+    AzureApi azureApi = samAzureApi(token);
+
+    try {
+      ActionManagedIdentityResponse resp =
+          SamRetry.retry(
+              () ->
+                  azureApi.getActionManagedIdentity(
+                      SamConstants.SamActionManagedIdentity.RESOURCE_TYPE,
+                      billingProfileId,
+                      SamConstants.SamActionManagedIdentity.ACTION));
+      return Optional.ofNullable(resp.getDisplayName());
```

Comment:
Should this return `resp.getObjectId()` instead of `getDisplayName()`? I'm guessing dependent code will want the Azure object ID of the UAMI.